apps/ex_nvr/lib/ex_nvr/recordings.ex

```diff
@@ -76,6 +76,39 @@ defmodule ExNVR.Recordings do
     )
   end
 
+  @spec delete_oldest_recordings(Device.t(), integer()) ::
+          {:ok, Ecto.Schema.t()} | {:error, Ecto.Changeset.t()}
+  def delete_oldest_recordings(device, limit) do
+    recordings =
+      Recording.oldest_recordings_by_device(device.id, limit)
+      |> Repo.all()
+
+    Multi.new()
+    |> Multi.run(:adapt_runs, fn _repo, _params ->
+      Enum.each(Enum.reverse(recordings), fn recording ->
+        Run.between_dates(recording.start_date, device.id)
+        |> Repo.one()
+        |> Map.put(:start_date, recording.end_date)
+        |> Repo.insert(on_conflict: :replace_all)
+      end)
```

Comment:
The problem with this approach:

- You probably request the same `run` multiple times
- You don't delete the old runs (a run with all the recordings deleted)

A better approach to this is as follows:
```elixir
Multi.new()
# delete all recordings
# get the oldest recording
# delete all the runs that have start date less than the start date of the oldest recording
# update the oldest run start date to match the start date of the oldest recording
# delete all the recording files